ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/map/MapReduce.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.map;
+
+import io.confluent.ksql.execution.codegen.helpers.TriFunction;
+import io.confluent.ksql.function.FunctionCategory;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import io.confluent.ksql.util.KsqlConstants;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * Reduce a map using an initial state and function
+ */
+@UdfDescription(
+    name = "map_reduce",
```

Comment:
can we name this `reduce_map`. Map Reduce has a different historical meaning that might make this confusing.

Reply:
renamed